code/modules/mob/living/simple_animal/hostile/illusion.dm

```diff
@@ -16,51 +16,47 @@
 	health = 100
 	speed = 0
 	faction = list(FACTION_ILLUSION)
-	var/life_span = INFINITY //how long until they despawn
-	var/mob/living/parent_mob
-	var/multiply_chance = 0 //if we multiply on hit
-	del_on_death = 1
+	del_on_death = TRUE
 	death_message = "vanishes into thin air! It was a fake!"
+	/// Weakref to what we're copying
+	var/datum/weakref/parent_mob_ref
+	/// Prob of getting a clone on attack
+	var/multiply_chance = 0
 
-
-/mob/living/simple_animal/hostile/illusion/Life(seconds_per_tick = SSMOBS_DT, times_fired)
-	..()
-	if(world.time > life_span)
-		death()
-
-
-/mob/living/simple_animal/hostile/illusion/proc/Copy_Parent(mob/living/original, life = 50, hp = 100, damage = 0, replicate = 0 )
+/mob/living/simple_animal/hostile/illusion/proc/Copy_Parent(mob/living/original, life = 5 SECONDS, hp = 100, damage = 0, replicate = 0)
 	appearance = original.appearance
-	parent_mob = original
+	parent_mob_ref = WEAKREF(original)
 	setDir(original.dir)
-	life_span = world.time+life
-	health = hp
+	maxHealth = hp
+	updatehealth()
 	melee_damage_lower = damage
 	melee_damage_upper = damage
 	multiply_chance = replicate
 	faction -= FACTION_NEUTRAL
 	transform = initial(transform)
 	pixel_x = base_pixel_x
 	pixel_y = base_pixel_y
-
+	addtimer(CALLBACK(src, TYPE_PROC_REF(/mob/living, death)), life)
 
 /mob/living/simple_animal/hostile/illusion/examine(mob/user)
+	var/mob/living/parent_mob = parent_mob_ref?.resolve()
 	if(parent_mob)
 		return parent_mob.examine(user)
-	else
-		return ..()
-
+	return ..()
 
 /mob/living/simple_animal/hostile/illusion/AttackingTarget()
 	. = ..()
-	if(. && isliving(target) && prob(multiply_chance))
-		var/mob/living/L = target
-		if(L.stat == DEAD)
+	if(!. || !isliving(target))
+		return
+	var/mob/living/parent_mob = parent_mob_ref?.resolve()
+	if(!isnull(parent_mob) && prob(multiply_chance))
```

Comment:
if you're refactoring this code to have more early returns can we also make this an early return too to prevent that needless+ugly indent